sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -174,17 +174,19 @@ class Dataset[T] private[sql](
     this(sqlContext.sparkSession, logicalPlan, encoder)
   }
 
-  @transient private[sql] val logicalPlan: LogicalPlan = {
+  @transient private val logicalPlan: LogicalPlan = {
```

Comment:
Exposing  the logical plan wrapped with the barrier tends to cause problem. So mark it as `private`.

We have `queryExecution.analyzed` to access the analyzed logical plan.